src/components/Filter/__tests__/useUrlFilters.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { NAME } from '_/utils/constants';
+
+import { act, cleanup, renderHook } from '@testing-library/react-hooks';
+
+import { useUrlFilters } from '../useUrlFilters';
+
+afterEach(cleanup);
+beforeEach(() => {
+  Object.defineProperty(window, 'location', {
+    writable: true,
+    value: { assign: jest.fn() },
+  });
+  window.location.pathname = '';
+  window.history.pushState = jest.fn();
+});
+
+describe('parse filters from the URL on initialization', () => {
+  const cases: [string, string, string[]][] = [
+    ['gets initialized with single value', '?name=%5B"foo"%5D', ['foo']],
+    ['gets initialized with 2 values', '?name=%5B"foo"%2C"bar"%5D', ['foo', 'bar']],
+    ['ignores empty JSON array', '?name=%5B%5D', undefined],
+    ['ignores data different then JSON array', '?name="foo"', undefined],
+    ['ignores corrupted data', '?name=%5B%foo', undefined],
+  ];
+
+  cases.forEach(([description, search, output]) =>
```

Comment:
Could also consider `it.each(cases)( ... )`

https://jestjs.io/docs/api#1-testeachtablename-fn-timeout